internal/controllers/cluster/cluster_controller_phases_test.go

```diff
@@ -119,6 +137,50 @@ func TestClusterReconcilePhases(t *testing.T) {
 				},
 				expectErr: false,
 			},
+			{
+				name:    "returns an error if the control plane endpoint is not yet set",
+				cluster: clusterNoEndpoint,
+				infraRef: map[string]interface{}{
+					"kind":       "GenericInfrastructureMachine",
+					"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+					"metadata": map[string]interface{}{
+						"name":              "test",
+						"namespace":         "test-namespace",
+						"deletionTimestamp": "sometime",
+					},
+					"status": map[string]interface{}{
+						"ready": true,
+					},
+				},
+				expectErr: true,
+			},
+			{
+				name:    "should propagate the control plane endpoint once set",
+				cluster: clusterNoEndpoint,
+				infraRef: map[string]interface{}{
+					"kind":       "GenericInfrastructureMachine",
+					"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+					"metadata": map[string]interface{}{
+						"name":              "test",
+						"namespace":         "test-namespace",
+						"deletionTimestamp": "sometime",
+					},
+					"spec": map[string]interface{}{
+						"controlPlaneEndpoint": map[string]interface{}{
+							"host": "example.com",
+							"port": int64(6443),
+						},
+					},
+					"status": map[string]interface{}{
+						"ready": true,
+					},
+				},
+				expectErr: false,
+				check: func(g *GomegaWithT, in *clusterv1.Cluster) {
+					g.Expect(in.Spec.ControlPlaneEndpoint.Host).To(Equal("example.com"))
+					g.Expect(in.Spec.ControlPlaneEndpoint.Port).To(BeEquivalentTo(6443))
+				},
```

Comment:
Aren't we missing a t.Run("reconcile control plane",  that runs reconcileControlPlane similar to what we do for infrastructure here?

Reply:
nvm I just saw it below